index.js

```diff
@@ -1,24 +1,79 @@
-function rot13(str) { // LBH QVQ VG!
-var shifted = 13;
-var splitted = str.split('');
-var arr = [];
-var inCharCode = splitted.forEach(function(element) {
-arr.push(element.charCodeAt(0));
-return arr;
-});
-var shifterArray = arr.map(function(element) {
-return element += shifted;
-})
-var someString = shifterArray.join( );
-var res = String.fromCharCode(someString);
-var resultArray = shifterArray.map(function (x) {
-return parseInt(x, 10);
-});
-var symbols = resultArray.forEach(function(val) {
-return String.fromCharCode(val);
-})
-return symbols;
+/**
+ * Used for tests, throws an error if value is not equal to expected value
+ */
+function assertEq(value, expected) {
+    if (value != expected) {
+        throw 'Expected ' + value + ' to be ' + expected;
+    }
 }
 
-// Change the inputs below to test
-rot13("SERR PBQR PNZC");
\ No newline at end of file
+/**
+ * @param {String} str Source string
+ * @param {Number} shift Encription key
+ * @param {Boolean} crypt If True, encrypt source string, otherwise decrypt it
+ * @param {String} [customAlphabet] Optional, real alphabet is used by default
+ * 
+ * @return {String} crypted or dectypted string
+ */
+function rot(str, shift, crypt, customAlphabet) {
+	// Custom alphabet is a sort of code book
+	// If letters would be placed in non-alphabetical order, receiver should know it to decrypt message
+	var alphabet = (customAlphabet || 'ABCDEFGHIJKLMNOPQRSTUVWXYZ').split(''),
+		len = alphabet.length;
+
+	// ABC (split)-> [A, B, C] (map)-> [B, C, D] (join)-> BCD
+	return str.split('').map(function(ch) {
+		var code = alphabet.indexOf(ch);
+
+		// In case symbol is not present in alphabet - just return it
+		if (code == -1) {
+			return ch;
+		}
+
+		if (crypt) {
+			code += shift; // Index of encrypted letter is {code} + {shift}
+		} else {
+			// Index of decrypted letter is {code} - {shift}
+			// We need to add length of the alphabet to this code for case when shift > code
+			code = (code - shift) + len;
```

Comment:
Remember that shift cannot be more than alphabet length.
Or you can do
shift = shift % len
after var at the beginning of the function